go/vt/vtgate/vcursor_impl.go

```diff
@@ -848,6 +848,12 @@ func (vc *vcursorImpl) SetPlannerVersion(v plancontext.PlannerVersion) {
 	vc.safeSession.GetOrCreateOptions().PlannerVersion = v
 }
 
+func (vc *vcursorImpl) SetPriority(priority string) {
+	if priority != "" {
+		vc.safeSession.GetOrCreateOptions().Priority = priority
+	}
+}
+
```

Comment:
we need to reset the priority here if any old value is present.

Reply:
I am unconditionally setting it now to the value passed in the string.